src/include/execution/sql/value.h

```diff
@@ -237,38 +179,54 @@ struct StringVal : public Val {
   std::string_view StringView() const {
     TERRIER_ASSERT(!is_null_,
                    "You should be doing a NULL check before attempting to generate a std::string_view of a StringVal.");
-    return std::string_view(Content(), len_);
+    return std::string_view(GetContent(), GetLength());
   }
 
   /**
-   * Helper method to turn a StringVal into a VarlenEntry.
-   * @param str input to be turned into a VarlenEntry
-   * @param own whether the varlen entry to own the string
-   * @return VarlenEntry representing StringVal
+   * @return Threshold for inlining.
+   */
+  static uint32_t InlineThreshold() { return storage::VarlenEntry::InlineThreshold(); }
+
+  /**
+   * Helper method to create a VarlenEntry from a StringVal.
+   * @param str The input to be turned into a VarlenEntry.
+   * @param own Whether the VarlenEntry should own the resulting string.
+   * @return A VarlenEntry with the contents of the StringVal.
    */
   static storage::VarlenEntry CreateVarlen(const StringVal &str, bool own) {
     if (str.is_null_) {
       return terrier::storage::VarlenEntry::CreateInline(static_cast<const terrier::byte *>(nullptr), 0);
```

Comment:
Is `VarlenEntry` meant to be able to represent a NULL? That seems inconsistent with the storage layer. What is the scenario when we would do this, and where is it defined that a `VarlenEntry` with a `nullptr` content and length 0 is considered SQL NULL?

Reply:
As I dig deeper into where it's used in the bytecode handler, it seems we would discard this value immediately anyway. The semantics here seem a bit wonky and we might want to chat about it in the lab.